src/spatialdata/dataloader/graph_loader.py

```diff
@@ -1,12 +1,29 @@
 from __future__ import annotations
 
-from typing import Literal
+from typing import Any, Literal
 
 import geopandas as gpd
 import numpy as np
 from scipy.spatial import KDTree
 
 from spatialdata._logging import logger
+from spatialdata._types import ArrayLike
+
+
+def _compute_percentile(tree: KDTree, percentile: int, centroids: ArrayLike) -> float:
+    logger.warning("computing percentile could take long for large datasets.")
+    dd = tree.sparse_distance_matrix(
+        tree,
+        max_distance=np.sqrt(
+            (np.max(centroids[:, 0]) - np.min(centroids[:, 0])) ** 2
+            + (np.max(centroids[:, 1]) - np.min(centroids[:, 1])) ** 2
+        ),
+    ).toarray()
+    np.fill_diagonal(dd, np.nan)
+    dd = dd[~np.isnan(dd)].reshape(dd.shape[0], dd.shape[1] - 1)
+    f = np.percentile(dd, percentile)
+    assert isinstance(f, float)
+    return f
 
```

Comment:
I moved this code in a function to avoid code duplication. There was one bug with a type (once centroids was a list of lists, once a np.array), probably due to code redundancy.